File: MYRAJOURNEY 2/MYRAJOURNEY/Models/RehabPlan.swift
import Foundation

struct RehabPlan: Codable, Identifiable {
    let id: String
    let title: String
    let description: String?
    let videoUrl: String?
    let exerciseName: String?
    let setsPerDay: Int?
    let repsPerSet: Int?
    let exercises: [RehabExercise]?
    
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        
        // Handle ID as Int or String
        if let intId = try? container.decode(Int.self, forKey: .id) {
            id = String(intId)
        } else {
            id = try container.decode(String.self, forKey: .id)
        }
        
        title = try container.decode(String.self, forKey: .title)
        description = try container.decodeIfPresent(String.self, forKey: .description)
        videoUrl = try container.decodeIfPresent(String.self, forKey: .videoUrl)
        exerciseName = try container.decodeIfPresent(String.self, forKey: .exerciseName)
        setsPerDay = try container.decodeIfPresent(Int.self, forKey: .setsPerDay)
        repsPerSet = try container.decodeIfPresent(Int.self, forKey: .repsPerSet)
        exercises = try container.decodeIfPresent([RehabExercise].self, forKey: .exercises)
    }
    
    struct RehabExercise: Codable, Identifiable {
        let id: Int
        let name: String
        let description: String?
        let sets: Int?
        let reps: String?
        let frequencyPerWeek: String?
        let completed: Bool
        
        enum CodingKeys: String, CodingKey {
            case id, name, description, sets, reps, completed
            case frequencyPerWeek = "frequency_per_week"
        }
    }
    
    enum CodingKeys: String, CodingKey {
        case id, title, description, exercises
        case videoUrl = "video_url"
        case exerciseName = "exercise_name"
        case setsPerDay = "sets_per_day"
        case repsPerSet = "reps_per_set"
    }
}

File: backend/backend-organized/myrajourney/data/model/RehabPlan.java
package com.example.myrajourney.data.model;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class RehabPlan {
    @SerializedName("id")
    private String id;

    @SerializedName("title")
    private String title;

    @SerializedName("description")
    private String description;

    @SerializedName("video_url")
    private String videoUrl;

    @SerializedName("exercise_name")
    private String exerciseName; // For single-exercise plans

    @SerializedName("sets_per_day")
    private Integer setsPerDay;

    @SerializedName("reps_per_set")
    private Integer repsPerSet;

    @SerializedName("exercises")
    private List<RehabExercise> exercises;

    // Inner class for individual exercises within a plan
    public static class RehabExercise {
        @SerializedName("id")
        private int id;

        @SerializedName("name")
        private String name;

        @SerializedName("description")
        private String description;

        @SerializedName("sets")
        private Integer sets;

        @SerializedName("reps")
        private String reps;

        @SerializedName("frequency_per_week")
        private String frequencyPerWeek;

        @SerializedName("completed")
        private boolean completed;

        public int getId() {
            return id;
        }

        public String getName() {
            return name;
        }

        public String getDescription() {
            return description;
        }

        public Integer getSets() {
            return sets;
        }

        public String getReps() {
            return reps;
        }

        public String getFrequencyPerWeek() {
            return frequencyPerWeek;
        }

        public boolean isCompleted() {
            return completed;
        }
    }

    // Getters
    public String getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getVideoUrl() {
        return videoUrl;
    }

    public String getExerciseName() {
        return exerciseName;
    }

    public Integer getSetsPerDay() {
        return setsPerDay;
    }

    public Integer getRepsPerSet() {
        return repsPerSet;
    }

    public List<RehabExercise> getExercises() {
        return exercises;
    }
}